src/app/large-display/large-display.component.ts

```diff
@@ -111,34 +153,107 @@ export class LargeDisplayComponent implements OnInit, AfterViewInit {
+  );
 
-  constructor(public timerService: TimerService) { }
+  readonly roundPath$ = new BehaviorSubject<RoundPath | null>(null);
+
+
+  constructor(
+    public timerService: TimerService,
+    public authService: AuthService,
+    public firebaseService: FirebaseService,
+    private zone: NgZone,
+  ) { }
 
   // TODO: These values are only here for testing. Eventually, we'll get this
   // information from the round service.
   public startTime = TimePeriod.fromMonthAndQuarter(4, 1);
   public endTime = TimePeriod.fromMonthAndQuarter(11, 4);
 
+  // TODO: For the moment, we're only using one fixed, preexisting round for
+  // all teachers. Eventually, teachers will each create their own sessions
+  // and rounds.
+  readonly demoRoundPath = {sessionId: 'demo-session', roundId: 'demo-round'};
+
   // The flowers displayed are essentially the demoFlowers at this moment
+
   ngOnInit() {
+    this.authService.logTeacherIn().then(() => {
+      // At time of this writing, AngularFire uses a DIY implementation of the
+      // Promise class that runs outside of the Angular Zone. As a result, code
+      // that runs inside this callback won't trigger change detection unless
+      // we tell it to.
+      // See https://github.com/google/google-api-javascript-client/issues/353
+      this.zone.run(() => {
+        this.currentScreen = ScreenId.Lobby;
+      });
+    });
+
+    // Link up observables so that the timer state gets sent to the current
+    // round in Firebase. (But don't do anything when the current round is
+    // null.)
+    combineLatest([this.roundPath$, this.timerService.running$]).pipe(
+      filter(([roundPath]) => roundPath !== null),
+    ).subscribe(([roundPath, running]) => {
+      this.firebaseService.updateRoundData(roundPath, {running});
+    });
+
+    combineLatest([this.roundPath$, this.timerService.currentTime$]).pipe(
+      filter(([roundPath]) => roundPath !== null),
+    ).subscribe(([roundPath, timePeriod]) => {
+      this.firebaseService.updateRoundData(roundPath, {
+        currentTime: timePeriod.time,
+      });
+    });
+  }
+
+  /**
+   * Create a new round within the session, set its initial state, and make .
+   *
+   * TODO: As of this iteration, this function just re-uses the old round and
+   * re-populates it with data.
+   */
+  startRound() {
+    this.currentScreen = ScreenId.DuringTheRound;
+
+    // Eventually, we'll create a new round, but for the moment, we'll just use
+    // this one.
+    this.roundPath$.next(this.demoRoundPath);
+    this.firebaseService.updateRoundData(this.demoRoundPath, {
+      flowerSpeciesIds: this.demoFlowerSpecies.map(species => species.id),
+    });
+
+    // Give the timer its starting state.
+    // (Because of the subscriptions we set up, these initial values should
+    // propagate to Firestore.)
     this.timerService.initialize({
       running: false,
       tickSpeed: 1000,
       currentTime: this.startTime,
       endTime: this.endTime
     });
```

Comment:
I feel like this functionality of "joining"/setting up a round should be in its own service, especially the subscribing to timer and pushing updates to Firebase. This would probably fit in the planned `TeacherRoundService` and some of it in the `TeacherSessionService` perhaps.